### ISSUE TITLE
Common System protection not working for m3u8 playlists 

### ISSUE THREAD

**Issue by nehaboob** [google/shaka-packager#760]
## System info
**Operating System**: Ubuntu 14.04 
**Shaka Packager Version**:  Generated with https://github.com/google/shaka-packager version v2.4.2-c60e988afa-release

## Issue and steps to reproduce the problem

packager \
'in=/media/h264_baseline_360p_600.mp4,stream=audio,init_segment=/media/audio/init.mp4,segment_template=/media/audio/$Number$.m4s,playlist_name=/media/audio/main.m3u8,hls_group_id=audio,hls_name=ENGLISH,drm_label=AUDIO' \
'in=/media/h264_baseline_360p_600.mp4,stream=video,init_segment=/media/h264_360p/init.mp4,segment_template=/media/h264_360p/$Number$.m4s,playlist_name=/media/h264_360p/main.m3u8,iframe_playlist_name=/media/h264_360p/iframe.m3u8,drm_label=SD' \
  --generate_static_live_mpd \
  --protection_systems CommonSystem \
  --protection_scheme cbcs \
  --enable_raw_key_encryption \
  --keys label=:key_id=f3c5e0361e6654b28f8049c778b23946:key=a4631a153a443df9eed0593043db7519\
  --hls_master_playlist_output /media/h264_master.m3u8 \
  --mpd_output /media/h264_master.mpd 

<Please provide any additional information below.>
I am using above command to encrypt media files for CommonSystem using cbcs protection scheme. 

Generated mpd plays fine in shaka-player, however when I tried playing m3u8 it gives error: hls_parser.js:1195 Unsupported HLS KEYFORMAT identity 
I am suspecting this is because of below key information in hls playlists.
#EXT-X-KEY:METHOD=SAMPLE-AES-CTR,URI="data:text/plain;base64,88XgNh5mVLKPgEnHeLI5Rg==",KEYFORMAT="identity"

Could you please point me to correct way of generating m3u8 playlists for CommonSystems which can be played by majority of players ?

**Comment by kqyang** [comment]
@nehaboob Can you confirm that you have specified "--protection_scheme cbcs"? The generated key format should be SAMPLE-AES with cbcs protection scheme.

**Comment by nehaboob** [comment]
@kqyang Sorry there was a typo. 

I am getting below EXT-X-KEY and its giving Unsupported HLS KEYFORMAT identity error in shaka player.

#EXT-X-KEY:METHOD=SAMPLE-AES,URI="data:text/plain;base64,88XgNh5mVLKPgEnHeLI5Rg==",IV=0xC0DF95C86FC0E3035960563362E491D0,KEYFORMAT="identity"

Packager command: 
packager \
'in=/media/h264_baseline_360p_600.mp4,stream=audio,init_segment=/media/audio/init.mp4,segment_template=/media/audio/$Number$.m4s,playlist_name=/media/audio/main.m3u8,hls_group_id=audio,hls_name=ENGLISH,drm_label=AUDIO' \
'in=/media/h264_baseline_360p_600.mp4,stream=video,init_segment=/media/h264_360p/init.mp4,segment_template=/media/h264_360p/$Number$.m4s,playlist_name=/media/h264_360p/main.m3u8,iframe_playlist_name=/media/h264_360p/iframe.m3u8,drm_label=SD' \
  --generate_static_live_mpd \
  --protection_systems CommonSystem \
  --protection_scheme cbcs \
  --enable_raw_key_encryption \
  --keys label=:key_id=f3c5e0361e6654b28f8049c778b23946:key=a4631a153a443df9eed0593043db7519\
  --hls_master_playlist_output /media/h264_master.m3u8 \
  --mpd_output /media/h264_master.mpd

**Comment by kqyang** [comment]
@nehaboob The packager output is correct. What operating system and what browser are you testing the playback on?

**Comment by nehaboob** [comment]
@kqyang I am testing on ubuntu 16.04 and Chrome Version 81.0

**Comment by kqyang** [comment]
@nehaboob You'll probably need to config the clear key explicitly in the player.

@joeyparrish Does Shaka Player support `identity` key format with keys provided in `URI`?

**Comment by joeyparrish** [comment]
Shaka Player does not currently support the identity key format in HLS.  To do so, we would likely use ClearKey, but Apple has not implemented that in Safari yet for some reason.

**Comment by joeyparrish** [comment]
It turns out we have an open feature request for that in Shaka Player: https://github.com/google/shaka-player/issues/2146

**Comment by kqyang** [comment]
@joeyparrish Thanks for the information.

**Comment by kqyang** [comment]
@nehaboob Let us know if there is anything else we can help with.

**Comment by nehaboob** [comment]
@kqyang I guess we just have to wait for clearkey to be support I guess.

**Comment by joeyparrish** [comment]
@nehaboob, you should file an issue on WebKit/Safari to ask for ClearKey to be supported.  If you do, please link to it here so we can track the issue as well.  Thanks!

**Comment by nehaboob** [comment]
@joeyparrish how about hls fmp4 clearkey support on Shaka player, is there any plan for that ?

**Comment by joeyparrish** [comment]
That is already supported.  You can use clearkey with any supported content on any browser with a clearkey implementation.
